### PR TITLE
Sticky pistons

### PR DESCRIPTION
**Pull request by WilliamHarmonMS**
I was asked for Sticky Pistons today, that they might be needed for mid tier levels, so I added some extra string parsing to allow for activatePistons / pushBlocks / deactivatePistons function calls to work with both sticky and normal pistons (plus a negligible amount of logic for the actual stickiness)

### CODE REVIEW

File: src/js/game/LevelMVC/LevelBlock.js [hunk 1]
```diff
@@ -144,7 +144,9 @@ module.exports = class LevelBlock {
     if (blockType.startsWith("piston")) {
       this.isDestroyable = false;
       this.isConnectedToRedstone = !blockType.startsWith("pistonArm");
-      if (blockType.substring(blockType.length - 2, blockType.length) === "On" || blockType.startsWith("pistonArm")) {
+      if (blockType.substring(blockType.length - 2, blockType.length) === "On" ||
+      blockType.startsWith("pistonArm") ||
+      blockType.substring(blockType.length - 8, blockType.length) === "OnSticky") {
```
Comment (Hamms) [inline src/js/game/LevelMVC/LevelBlock.js]: nit: multiline control statements [should be indented on their own lines](https://github.com/airbnb/javascript#control-statements):

```javascript
if (
  blockType.substring(blockType.length - 2, blockType.length) === "On" ||
  blockType.startsWith("pistonArm") ||
  blockType.substring(blockType.length - 8, blockType.length) === "OnSticky"
) {
```

File: src/js/game/LevelMVC/LevelPlane.js [hunk 2]
```diff
@@ -418,6 +418,10 @@ module.exports = class LevelPlane {
     let pos = [];
     let offset = [];
     let pistonType = this.getBlockAt(position).blockType;
+    let checkSticky = pistonType.substring(pistonType.length - 6, pistonType.length);
+    if (checkSticky === "Sticky") {
```
Comment (Hamms) [inline src/js/game/LevelMVC/LevelPlane.js]: this (and the several instances below) should be in a `levelBlock.getIsStickyPiston()` method

File: src/js/game/LevelMVC/LevelPlane.js [hunk 3]
```diff
@@ -479,7 +492,11 @@ module.exports = class LevelPlane {
     let east = 2;
     let west = 3;
 
-    switch (this.getBlockAt(position).blockType) {
+    let pistonType = this._data[this.coordinatesToIndex(position)].blockType;
+    if (pistonType.substring(pistonType.length - 6, pistonType.length) === "Sticky") {
+      pistonType = pistonType.substring(0, pistonType.length - 6);
+    }
```
Comment (Hamms) [inline src/js/game/LevelMVC/LevelPlane.js]: please continue to use `getBlockAt` rather than `_data`

File: src/js/game/LevelMVC/LevelPlane.js [hunk 4]
```diff
@@ -534,11 +586,24 @@ module.exports = class LevelPlane {
     let armBlock = new LevelBlock(pistonType);
     for (let i = blocksPositions.length - 1; i >= 0; --i) {
       let destination = [blocksPositions[i][0] + offsetX, blocksPositions[i][1] + offsetY];
-      this.setBlockAt(destination, this.getBlockAt(blocksPositions[i]));
+      let block = this.getBlockAt(blocksPositions[i]);
+      if (this.getBlockAt(destination) === undefined) {
+        console.log("break");
+      }
```
Comment (Hamms) [inline src/js/game/LevelMVC/LevelPlane.js]: nit: console.log

**Comment by WilliamHarmonMS** [comment]
@Hamms PTAL 👍

**Pull request merged**